Code/Framework/AzCore/AzCore/Serialization/Json/JsonDeserializer.cpp

```diff
@@ -571,9 +571,23 @@ namespace AZ
             if (loadedTypeId.m_determination == TypeIdDetermination::FailedToDetermine ||
                 loadedTypeId.m_determination == TypeIdDetermination::FailedDueToMultipleTypeIds)
             {
-                AZStd::string_view message = loadedTypeId.m_determination == TypeIdDetermination::FailedDueToMultipleTypeIds ?
-                    "Unable to resolve provided type because the same name points to multiple types." :
-                    "Unable to resolve provided type.";
+                AZStd::string message;
+                if (loadedTypeId.m_determination == TypeIdDetermination::FailedDueToMultipleTypeIds)
+                {
+                    message = "Unable to resolve provided type because the same name points to multiple types.";
+                }
+                else
+                {
+                    auto typeField = pointerData.FindMember(JsonSerialization::TypeIdFieldIdentifier);
+                    if (typeField != pointerData.MemberEnd() && typeField->value.IsString())
+                    {
+                        message = AZStd::string::format("Unable to resolve provided type: %s.", typeField->value.GetString());
+                    }
+                    else
+                    {
+                        message = "Unable to resolve provided type.";
+                    }
+                }
                 status = context.Report(Tasks::RetrieveInfo, Outcomes::Unknown, message);
                 return ResolvePointerResult::FullyProcessed;
```

Comment:
```suggestion
                    auto typeField = pointerData.FindMember(JsonSerialization::TypeIdFieldIdentifier);
                    if (typeField != pointerData.MemberEnd() && typeField->value.IsString())
                    {
                        const char* format = loadedTypeId.m_determination == TypeIdDetermination::FailedToDetermine ?
                            "Unable to resolve provided type: %.*s." : 
                            "Unable to resolve provided type %.*s because the same name points to multiple types.";
                        status = context.Report(Tasks::RetrieveInfo, Outcomes::Unknown, 
                            AZStd::string::format("Unable to resolve provided type: %.*s.", typeField->value.GetStringLength(), typeField->value.GetString()));
                    }
                    else
                    {
                        const char* message = loadedTypeId.m_determination == TypeIdDetermination::FailedToDetermine ?
                            "Unable to resolve provided type." :
                            "Unable to resolve provided type because the same name points to multiple types."
                        status = context.Report(Tasks::RetrieveInfo, Outcomes::Unknown, "Unable to resolve provided type.");
                    }
                    return ResolvePointerResult::FullyProcessed;
```
This shows the type id as well for case where names conflict, it avoids `AZStd::string` when it's not needed to avoid the memory allocation and it uses the string length for the type id.